Source/DafnyLanguageServer/Workspace/FileVerificationStatus.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MediatR;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Workspace;
+
+public record FileVerificationStatus(
+  DocumentUri Uri,
+  int? Version,
+  IReadOnlyList<NamedVerifiableStatus> NamedVerifiables) : IRequest {
+
+  public virtual bool Equals(FileVerificationStatus? other) {
+    if (ReferenceEquals(null, other)) {
+      return false;
+    }
+
+    if (ReferenceEquals(this, other)) {
+      return true;
+    }
+
+    return Uri.Equals(other.Uri) && Version == other.Version && NamedVerifiables.SequenceEqual(other.NamedVerifiables);
+  }
+
+  public override int GetHashCode() {
+    return HashCode.Combine(Uri, Version, NamedVerifiables);
+  }
+}
+
+/**
+ * Groups Boogie verification tasks by named Dafny declarations such as:
+ * methods,
+ * functions,
+ * data-types (whose default-value expressions are verified),
+ * fields (whose initial value is verified)
+ * types definitions (for example the verification of a witness of a subset type)
+ */
+public record NamedVerifiableStatus(Range NameRange, PublishedVerificationStatus Status) {
+  public virtual bool Equals(NamedVerifiableStatus? other) {
+    if (ReferenceEquals(null, other)) {
+      return false;
+    }
+
+    if (ReferenceEquals(this, other)) {
+      return true;
+    }
+
+    return NameRange.Equals(other.NameRange) && Status == other.Status;
+  }
+
+  public override int GetHashCode() {
+    return HashCode.Combine(NameRange, (int)Status);
+  }
+}
+
+public enum PublishedVerificationStatus {
+  Stale,    // Not scheduled to be run
+  Queued,   // Scheduled to be run but waiting for resources
+  Running,  // Currently running
```

Comment:
There is no need to expose anything below the "verifiable" level in the UI with my suggestion.
You'd  just need to adapt the function "combine(status, status)" so that

```
combine(Stale, Running) = Running
combine(Queued, Running) = Running
combine(Error, Running) = RunningWithError
combine(Verified, Running) = Running
combine(_, RunningWithError) = RunningWithError

combine(_, Error) = Error
....
```

Since there is no child display and you have the guarantee that an error in an implementation will result in an error in the method at the end, you shouldn't delay that information, so if you don't want to specify `RunningWithError`, I think that you should still have `Error` override `Running`. You could always display to the user the same for `Error` and `RunningWithError` right now if you are concerned that users will look for children.

If you do not add `RunningWithError` now, it will be possible to add `RunningWithError` later in this enum in a separate PR, but since this enum will already have become public API, we would have to add it at the end, which I don't think looks right.

Reply:
> We can skip an index between Running and Error, then we can later insert RunningWithError without penalty. Does that work?

That sounds like a good consensus, thanks